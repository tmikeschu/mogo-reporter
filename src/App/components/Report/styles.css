.Report .grid__title {
  font-size: 2em;
}

.Report .grid__container {
  display: grid;
  grid-template-columns: 30% 25% 25% 25%;
  background-color: transparent;
  align-items: center;
}
