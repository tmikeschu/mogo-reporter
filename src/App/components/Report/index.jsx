import React from "react";
import "./styles.css";
import EmojiInput from "../EmojiInput/";

const report = {
  danger: [{ name: "obstacle", emoji: "🕳️" }, { name: "car", emoji: "🚘" }],
  level: [
    { name: "high", emoji: "😨" },
    { name: "medium", emoji: "😵" },
    { name: "low", emoji: "😒" },
  ],
};

const gridFactory = (part, key, props) => {
  const parts = {
    grid: (t, i) => {
      let cell = gridFactory("cell", key, props);
      return cell(t, i);
    },
    cell: (t, i) => (
      <EmojiInput
        input={t}
        parent={key}
        selected={props[key]}
        onClick={props.onClick}
        key={`emojiInput-${t["name"]}`}
      />
    ),
  };
  return parts[part];
};

const Report = props => {
  return (
    <form className="Report">
      {Object.keys(report).map(key => {
        return (
          <span className={"grid__container"} key={key}>
            <p className="grid__title">{key}</p>
            {report[key].map((t, i) => {
              let grid = gridFactory("grid", key, props);
              return grid(t, i);
            })}
          </span>
        );
      })}
    </form>
  );
};

export default Report;
