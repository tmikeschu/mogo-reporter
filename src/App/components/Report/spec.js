import React from "react";
import { mount } from "enzyme";
import Report from "./";

describe("<Report />", () => {
  const wrapper = shallow(<Report />);
  const container = wrapper.instance();

  it("mounts without crashing", () => {
    const wrapper = mount(<Report />);
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
