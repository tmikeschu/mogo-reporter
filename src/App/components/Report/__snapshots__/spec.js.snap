// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Report /> mounts without crashing 1`] = `
<Report>
  <form
    className="Report"
  >
    <span
      className="grid__container"
      key="danger"
    >
      <p
        className="grid__title"
      >
        danger
      </p>
      <EmojiInput
        input={
          Object {
            "emoji": "🕳️",
            "name": "obstacle",
          }
        }
        key="emojiInput-obstacle"
        parent="danger"
      >
        <span
          className="emoji"
        >
          <label
            className="emoji--selected-false"
            htmlFor="obstacle"
          >
            <span>
              🕳️
            </span>
          </label>
          <input
            checked={false}
            data-metric="danger"
            id="obstacle"
            type="radio"
            value="obstacle"
          />
        </span>
      </EmojiInput>
      <EmojiInput
        input={
          Object {
            "emoji": "🚘",
            "name": "car",
          }
        }
        key="emojiInput-car"
        parent="danger"
      >
        <span
          className="emoji"
        >
          <label
            className="emoji--selected-false"
            htmlFor="car"
          >
            <span>
              🚘
            </span>
          </label>
          <input
            checked={false}
            data-metric="danger"
            id="car"
            type="radio"
            value="car"
          />
        </span>
      </EmojiInput>
    </span>
    <span
      className="grid__container"
      key="level"
    >
      <p
        className="grid__title"
      >
        level
      </p>
      <EmojiInput
        input={
          Object {
            "emoji": "😨",
            "name": "high",
          }
        }
        key="emojiInput-high"
        parent="level"
      >
        <span
          className="emoji"
        >
          <label
            className="emoji--selected-false"
            htmlFor="high"
          >
            <span>
              😨
            </span>
          </label>
          <input
            checked={false}
            data-metric="level"
            id="high"
            type="radio"
            value="high"
          />
        </span>
      </EmojiInput>
      <EmojiInput
        input={
          Object {
            "emoji": "😵",
            "name": "medium",
          }
        }
        key="emojiInput-medium"
        parent="level"
      >
        <span
          className="emoji"
        >
          <label
            className="emoji--selected-false"
            htmlFor="medium"
          >
            <span>
              😵
            </span>
          </label>
          <input
            checked={false}
            data-metric="level"
            id="medium"
            type="radio"
            value="medium"
          />
        </span>
      </EmojiInput>
      <EmojiInput
        input={
          Object {
            "emoji": "😒",
            "name": "low",
          }
        }
        key="emojiInput-low"
        parent="level"
      >
        <span
          className="emoji"
        >
          <label
            className="emoji--selected-false"
            htmlFor="low"
          >
            <span>
              😒
            </span>
          </label>
          <input
            checked={false}
            data-metric="level"
            id="low"
            type="radio"
            value="low"
          />
        </span>
      </EmojiInput>
    </span>
  </form>
</Report>
`;
